### PR TITLE
feat(ui): login & sign-up page for IDPE

### PR DESCRIPTION
**Pull request by Asalem1**
This PR is the final part in a series of PRs geared at integrating a new sign-up/ log-in flow for cloud users. The view of this page is supposed to mirror the refactored version that exists as a branch on Quartz (and will eventually become the main log-in & sign-up page for Cloud2). This PR was broken down into a separate branch and worked upon in the following PRs:

- Add routing logic (https://github.com/influxdata/influxdb/pull/16979)
- Added Auth0 and Basic form functionality (https://github.com/influxdata/influxdb/pull/16987)
- Created a Social Button component for sign-in through social identity provider (https://github.com/influxdata/influxdb/pull/17033)
- Added `react-spring` for transition effects between form toggles (https://github.com/influxdata/influxdb/pull/17044)
- Added logout functionality (https://github.com/influxdata/influxdb/pull/17045)

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)

### CODE REVIEW

**Comment by Asalem1** [comment]
Once this feature gets merged, we can start the process of rolling out the new config variables:

https://github.com/influxdata/k8s-idpe/pull/444

File: ui/src/onboarding/containers/LoginPage.scss [hunk 1]
```diff
@@ -0,0 +1,166 @@
+.clockface--app-wrapper.sign-up--page {
+  background-image: linear-gradient(
+    47deg,
+    rgba(19, 0, 45, 0) 65%,
+    rgba(191, 47, 229, 0.4) 100%
+  ) !important;
+  background-color: #13002d !important;
+}
```
Comment (hoorayimhelping) [inline ui/src/onboarding/containers/LoginPage.scss]: might be worth doing `div.clockface--app-wrapper`, may not be possible if more than divs are used

File: ui/src/onboarding/containers/LoginPageContents.tsx [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+    const isValid = !Object.values(errors).some(error => !!error)
+
+    return {isValid, errors}
+  }
+
+  private formFieldTypeFactory = (
+    errorMessage: string
+  ): FormFieldValidation => ({
+    errorMessage,
+    isValid: errorMessage !== '',
```
Comment (Asalem1) [inline ui/src/onboarding/containers/LoginPageContents.tsx]: isValid should be `errorMessage === ''`

File: ui/src/onboarding/containers/LoginPageContents.tsx [hunk 3]
```diff
@@ -0,0 +1,443 @@
...
+              enter={[
+                {
+                  position: 'relative',
+                  overflow: 'hidden',
+                  height: 'auto',
+                },
+              ]}
+              leave={{height: 0}}
+            >
+              {show =>
```
Comment (Asalem1) [inline ui/src/onboarding/containers/LoginPageContents.tsx]: change `show` to `shouldShow`

File: ui/src/onboarding/containers/LoginPageContents.tsx [hunk 4]
```diff
@@ -0,0 +1,443 @@
...
```
Comment (Asalem1) [inline ui/src/onboarding/containers/LoginPageContents.tsx]: see below

File: ui/src/onboarding/containers/LoginPageContents.tsx [hunk 5]
```diff
@@ -0,0 +1,443 @@
...
+      emailError,
+      passwordError,
+    }
+    if (activeTab === 'signup') {
+      errors.firstNameError = firstNameError
+      errors.lastNameError = lastNameError
+      errors.confirmPasswordError = confirmPasswordError
+    }
+
+    const isValid = !Object.values(errors).some(error => !!error)
```
Comment (Asalem1) [inline ui/src/onboarding/containers/LoginPageContents.tsx]: change `some` to `every` to check for an empty string

Comment (Asalem1) [reply ui/src/onboarding/containers/LoginPageContents.tsx]: change `isValid` to actually mean that there are no errors, right now it's indicating that there are errors

File: ui/src/onboarding/containers/LoginPageContents.tsx [hunk 6]
```diff
@@ -0,0 +1,443 @@
...
+        family_name,
+        given_name,
+      },
+      error => {
+        if (error) {
+          this.displayErrorMessage(errors, error)
+          this.setState({buttonStatus: ComponentStatus.Default})
+          return
+        }
+        // log the user into Quartz
```
Comment (Asalem1) [inline ui/src/onboarding/containers/LoginPageContents.tsx]: login to auth0, not Quartz

File: ui/src/onboarding/containers/LoginPageContents.tsx [hunk 7]
```diff
@@ -0,0 +1,443 @@
...
+              return
+            }
+          }
+        )
+      }
+    )
+  }
+
+  private displayErrorMessage = (errors, auth0Err) => {
+    if (auth0Err.code.includes('error in email')) {
```
Comment (Asalem1) [inline ui/src/onboarding/containers/LoginPageContents.tsx]: switch this back to `regex` from Quartz

File: ui/src/onboarding/containers/LoginPageContents.tsx [hunk 8]
```diff
@@ -0,0 +1,443 @@
...
+    } else if (auth0Err.code === 'user_exists') {
+      const emailError = `An account with that email address already exists. Try logging in instead.`
+      this.setState({...errors, emailError})
+    } else {
+      const emailError = `We have been notified of an issue while creating your account. If this issue persists, please contact support@influxdata.com`
+      this.setState({...errors, emailError})
+    }
+  }
+
+  private handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
```
Comment (Asalem1) [inline ui/src/onboarding/containers/LoginPageContents.tsx]: change `e` to `event`

File: ui/src/onboarding/containers/LoginPageContents.tsx [hunk 9]
```diff
@@ -0,0 +1,458 @@
...
+              return
+            }
+          }
+        )
+      }
+    )
+  }
+
+  private displayErrorMessage = (errors, auth0Err) => {
+    if (/error in email/.test(auth0Err.code)) {
```
Comment (hoorayimhelping) [inline ui/src/onboarding/containers/LoginPageContents.tsx]: 👍

**Pull request merged**